Sources/SpeziLLMLocal/LLMLocalSession.swift

```diff
@@ -113,13 +115,45 @@ public final class LLMLocalSession: LLMSession, @unchecked Sendable {
         }
     }
     
+    /// Releases the memory associated with the current model.
+    ///
+    /// This function frees up memory resources by clearing the model container and reset the GPU cache, allowing to e.g. load a different model.
+    public func offload() async {
+        cancel()
+        await MainActor.run {
+            modelContainer = nil
+            state = .uninitialized
+        }
+        MLX.GPU.clearCache()
+    }
+    
     
     /// Based on the input prompt, generate the output.
-    /// - Returns: A Swift `AsyncThrowingStream` that streams the generated output.
+    /// - Returns: A Swift `AsyncThrowingStream` that streams the generated output as `String`.
     @discardableResult
     public func generate() async throws -> AsyncThrowingStream<String, Error> {
         let (stream, continuation) = AsyncThrowingStream.makeStream(of: String.self)
         
+        async let generate: AsyncThrowingStream<LLMLocalGenerateState, Error> = generate()
+        
+        for try await state in try await generate {
```

Comment:
Wondering if we could use a map/compactmap or something similar here instead of a continuation and for loop. Also applies to the code below.